### PR TITLE
crimson/seastore: introduce and adopt LBAManager::get_mapping(t, offset)

### PR DESCRIPTION
**Pull request by cyx1231st**
Should be no impact to existing users.

This is a prerequisite PR that allows us to read an extent without knowing its length first, using the new `get_mapping()` method.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc [hunk 1]
```diff
@@ -107,7 +107,17 @@ BtreeLBAManager::get_mapping(
   Transaction &t,
   laddr_t offset)
 {
-  ceph_abort("not implemented");
+  logger().debug("BtreeLBAManager::get_mapping: {}", offset);
+  return get_root(t
+  ).safe_then([this, &t, offset](auto extent) {
+    return extent->lookup_pin(get_context(t), offset
+    ).safe_then([extent] (auto ret) { return ret; });
```
Comment (tchaikov) [inline src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc:114]: I'd suggest flatten this continuation chain, i.e., to promote the continuation above to the outer chain.

Comment (tchaikov) [reply src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc:114]: why is `extent` captured?

**Review by tchaikov — APPROVED** [review]

File: src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc (hunk 1 above)
Comment (cyx1231st) [reply src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc:114]: Hmm, seems no reason to keep the `extent` reference as `lookup_pin()` already handles it well internally.

Comment (cyx1231st) [reply src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc]: fixed.

File: src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc [hunk 2]
```diff
@@ -107,7 +107,16 @@ BtreeLBAManager::get_mapping(
   Transaction &t,
   laddr_t offset)
 {
-  ceph_abort("not implemented");
+  logger().debug("BtreeLBAManager::get_mapping: {}", offset);
+  return get_root(t
+  ).safe_then([this, &t, offset](auto extent) {
```
Comment (tchaikov) [inline src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc:112]: nit, might want to add a space between `]` and `(`. to be more consistent.

Comment (cyx1231st) [reply src/crimson/os/seastore/lba_manager/btree/btree_lba_manager.cc]: fixed.

**Review by tchaikov — APPROVED** [review]

**Pull request merged**